### PR TITLE
DHFPROD-7874: Display records related via foreign key in graph view

### PR DESCRIPTION
**Pull request by xnikhil08**
### Description
Some of the e2e test cases will be added separately, as part of DHFPROD-8208.
This involves changes for the ticket DHFPROD-7875 as well.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs [hunk 1]
```diff
@@ -128,60 +128,78 @@ let edges = [];
...
-      nodes[item.firstObjectIRI] = objectNode;
+      nodes[objectId] = objectNode;
     }
   }
   else if (item.predicateIRI !== undefined && item.predicateIRI.toString().length > 0){
       let edge = {};
 
       let predicateArr = item.predicateIRI.toString().split("/");
       let edgeLabel = predicateArr[predicateArr.length - 1];
+      edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI;
```
Comment (aniamy) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs:200]: Found a small issue with repeated ids on some edges. Could you push the following fix?
Full line should be:
` edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI + "-" + item.objectIRI;`

**Comment by xnikhil08** [comment]
> Edges are cossing over other edges. Should we handle that issue in another story or is part of this work? image

@aniamy - That will be handled as part of the story DHFPROD-8215, where we will take care of physics nuances along with preserving the last view of explorer.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs (hunk 1 above)
Comment (xnikhil08) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs]: Done.

**Review by aniamy — DISMISSED** [review]

**Review by Sanjeevani19 — APPROVED** [review]
UI Looks good !! 
One of the core unit test is failing, please look into that https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-6351/9/tests

**Comment by prolonMarkLogic** [comment]
@Sanjeevani19  I uploaded a fix for this core unit test

**Review by aniamy — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**